### ISSUE TITLE
mcr.microsoft.com is down

### ISSUE THREAD

**Issue by wbrefvem** [microsoft/containerregistry#62]
I'm sure this isn't the right place to report this, but mcr.microsoft.com is down. Is there a way to check its status and/or report an error? `status.azure.com` has an entry for "Container Registry" but it's not clear if this is the same thing. Any guidance you could provide would be appreciated.

Here's what goes wrong:

```
$ docker pull mcr.microsoft.com/dotnet/core/sdk:3.1
Error response from daemon: Get https://mcr.microsoft.com/v2/: Service Unavailable
```

I got this error running locally on macOS. Restarting the docker daemon had no effect. Pulling the same image in a docker build running on GitHub Actions produces the same error.

**Comment by wbrefvem** [comment]
Actually the error on GitHub Actions is perhaps more informative:

```
Get https://mcr.microsoft.com/v2/: dial tcp 255.255.255.255:443: connect: network is unreachable
```

It seems that DNS records for `mcr.microsoft.com` are incorrect.

Locally, on macOS:

```
$ dig mcr.microsoft.com

; <<>> DiG 9.10.6 <<>> mcr.microsoft.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 9939
;; flags: qr rd ra; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4000
;; QUESTION SECTION:
;mcr.microsoft.com.		IN	A

;; ANSWER SECTION:
mcr.microsoft.com.	1601	IN	CNAME	global.fe.mscr.io.
global.fe.mscr.io.	111	IN	CNAME	mcr-global.trafficmanager.net.
mcr-global.trafficmanager.net. 28 IN	CNAME	rpm0408eus.eastus.cloudapp.azure.com.
rpm0408eus.eastus.cloudapp.azure.com. 10 IN A	255.255.255.255

;; Query time: 50 msec
;; SERVER: <redacted>#53(<redacted>)
;; WHEN: Thu May 27 16:56:27 EDT 2021
;; MSG SIZE  rcvd: 183
```

**Comment by doominator42** [comment]
I'm having the exact same issue as well.
I tested from Google and Cloudflare DNS servers and both from my home network and from an OVH server. They all returns `255.255.255.255` for `rpm0408eus.eastus.cloudapp.azure.com`.

`mcr-global.trafficmanager.net` seems to randomly alternate between `rpm0408eus.eastus.cloudapp.azure.com` and `rpm0507eus.eastus.cloudapp.azure.com` causing intermittent interruptions.

**Comment by kkorniszuk** [comment]
I got the same with containerized .net app on heroku

**Comment by ctmdev21** [comment]
msft rescue our CD pipelines :fire::fire::fire:

**Comment by leojonathanoh** [comment]
github worfklows are failing with the same DNS issue, can't pull any powershell images. https://github.com/theohbrothers/PSSiteScraper/pull/29/checks?check_run_id=2689382257

**Comment by toddysm** [comment]
Looking into it.

**Comment by wbrefvem** [comment]
> `mcr-global.trafficmanager.net` seems to randomly alternate between `rpm0408eus.eastus.cloudapp.azure.com` and `rpm0507eus.eastus.cloudapp.azure.com` causing intermittent interruptions.

I've noticed the same thing. It could be an issue with local DNS caching? Not sure what to do about that on GitHub.

**Comment by wbrefvem** [comment]
It looks like this may have been a temporary DNS issue -- I seem to be getting `rpm0507eus.eastus.cloudapp.azure.com` pretty consistently in the chain of CNAME records attached to `mcr.microsoft.com`, and that seems to resolve to a valid IP. But I'm going to leave this issue open until some from Microsoft weighs in.

**Comment by leojonathanoh** [comment]
> github worfklows are failing with the same DNS issue, can't pull any powershell images. https://github.com/theohbrothers/PSSiteScraper/pull/29/checks?check_run_id=2689381991
> 
> EDIT: It appears that some ci jobs are failing but some are fine. If this is a DNS issue perhaps its related to propagation or caching.

appears to have resolved

**Comment by toddysm** [comment]
@wbrefvem and @leojonathanoh We have mitigated the issue. You may see some occasional failures because of DNS caching. I will post more details what the issue was once we do the post mortem.

**Comment by wbrefvem** [comment]
@toddysm thanks!

**Comment by toddysm** [comment]
**Root Cause:** During our investigation, we identified a networking issue caused by a recent deployment that altered MCR's DNS configuration. In this case, MCR's DNS was affected at one point of ingestion which resolves requests for multiple regions. This resulted in MCR being intermittently unavailable to process the service management requests.

**Mitigation:** We rerouted traffic for MCR request to a healthy region, where we were able to validate all pending requests were successfully completed, mitigated the issue.
